tests/test_gui.py

```diff
@@ -63,17 +64,15 @@ def host_application(context):
             context["type"],
             "--context-entity-id",
             str(context["id"]),
-        ]
+        ],
+        stdout=subprocess.PIPE,
```

Comment:
I don't understand this change.
Also, I don't understand the previous code. If we don't pass any stdout/stderr to Popen, then the child process's stdout/stderr will be inherited from the parent. Which means it will be sent to sys.stdout. So the finally block in the previous code did not make sense.

Also, in the new code, why do we need a pipe? Don't we want to keep the same behavior? And also, pipe come with buffer limit so it's usually not a good idea if there is not something (a thread) consuming it periodically.

Reply:
This was just to try running the tests locally. I'll revert the changes on this file so we can work on this on the appropriate ticket.